### PR TITLE
feat: plugins

### LINKED ISSUE #15: Plugin System

### PR DESCRIPTION
**Pull request by UnseenFaith**
Fixes #15

### CODE REVIEW

File: src/client/BaseClient.ts [hunk 1]
```diff
@@ -5,6 +5,12 @@ import { TimerManager } from '@klasa/timer-manager';
 
 import { BaseClientOptionsDefaults } from '../util/Constants';
 
+const plugins: Set<Function> = new Set();
```
Comment (kyranet) [inline src/client/BaseClient.ts:8]: Alternatively, you can do this:
```suggestion
const plugins = new Set<Function>();
```

Comment (bdistin) [reply src/client/BaseClient.ts:8]: plugins should be a true private static property of the client. I am also unsure if this pr belongs on BaseClient or Client. Others chime in with arguments for either.

Comment (bdistin) [reply src/client/BaseClient.ts:8]: Actually, I am pretty sure it has to be on Client because of Piece Store access.

File: src/client/Client.ts [hunk 2]
```diff
@@ -8,6 +8,10 @@ import { ClientOptionsDefaults } from '../util/Constants';
 import type { Store } from '../lib/structures/base/Store';
 import type { Piece } from '../lib/structures/base/Piece';
 
+export interface Plugin {
+	[Client.plugin]: Function;
```
Comment (vladfrangu) [inline src/client/Client.ts:12]: Can't we type the function? (might me able to make it know that `this` is the Client instance

Comment (vladfrangu) [reply src/client/Client.ts:12]: The answer was: Unnecessary.

**Pull request merged**